akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java

```diff
@@ -32,78 +35,147 @@
  *
  * This is an EXPERIMENTAL feature and is subject to change until it has received more real world testing.
  */
-public class ReceiveBuilder {
+public class ReceiveBuilder extends AbstractPFBuilder<Object, BoxedUnit> {
+
   private ReceiveBuilder() {
   }
 
   /**
-   * Return a new {@link UnitPFBuilder} with no case statements. They can be added later as the returned {@link
-   * UnitPFBuilder} is a mutable object.
+   * Return a new {@link ReceiveBuilder} with no case statements. They can be
+   * added later as the returned {@link ReceiveBuilder} is a mutable object.
    *
    * @return a builder with no case statements
    */
-  public static UnitPFBuilder<Object> create() {
-    return new UnitPFBuilder<>();
+  public static ReceiveBuilder create() {
+    return new ReceiveBuilder();
   }
 
+  // FIXME
+  //  public static PartialFunction<Object, BoxedUnit> onMessage(FI.UnitApply<Object> apply) {
+  //    return matchAny(apply).build();
+  //  }
+
   /**
-   * Return a new {@link UnitPFBuilder} with a case statement added.
+   * Add a new case statement to this builder.
    *
-   * @param type  a type to match the argument against
-   * @param apply an action to apply to the argument if the type matches
+   * @param type
+   *          a type to match the argument against
+   * @param apply
+   *          an action to apply to the argument if the type matches
    * @return a builder with the case statement added
    */
-  public static <P> UnitPFBuilder<Object> match(final Class<? extends P> type, FI.UnitApply<? extends P> apply) {
-    return UnitMatch.match(type, apply);
+  @SuppressWarnings("unchecked")
+  public <P> ReceiveBuilder match(final Class<? extends P> type, final FI.UnitApply<? extends P> apply) {
```

Comment:
Thanks for your insights. 
`match(List.class, (List<String> msg) -> ...)` doesn't sound like a good thing to do.
I think we should fix it as you suggest.

There are also other builders (not `ReceiveBuilder`) that are used in stable APIs such as Akka Streams `collect`. I'm more worried about changing those and break binary compatibility of Akka Streams. Have to look into that part.